File: 02_ProgrammingFundamentals/_09_MapsLambdaAndStreamAPI/_01_Lecture/_03_OddOccurrences.java
package _02_ProgrammingFundamentals._09_MapsLambdaAndStreamAPI._01_Lecture;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.Scanner;

public class _03_OddOccurrences {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        String[] words = scan.nextLine().split(" ");

        // Add variable
        LinkedHashMap<String, Integer> counts = new LinkedHashMap<>();

        // Extract the elements that occur odd times
        for (String word : words) {
            String wordInLowerCase = word.toLowerCase();

            if (counts.containsKey(wordInLowerCase)) {
                counts.put(wordInLowerCase, counts.get(wordInLowerCase) + 1);
            } else {
                counts.put(wordInLowerCase, 1);
            }
        }

        ArrayList<String> odds = new ArrayList<>();

        for (var entry : counts.entrySet()) {
            if (entry.getValue() % 2 == 1) {
                odds.add(entry.getKey());
            }
        }

        // Print result
        for (int i = 0; i < odds.size(); i++) {
            System.out.print(odds.get(i));

            if (i < odds.size() - 1) {
                System.out.print(", ");
            }
        }

        scan.close();
    }
}

File: 24_C#Fundamentals/_07_AssociativeArrays/_01_Lecture/_03_OddOccurrences/_03_OddOccurrences.cs
using System;

public class _03_OddOccurrences
{
    public static void Main(string[] args)
    {
        // Read input values
        string[] words = Console.ReadLine().Split(" ");

        // Add variable
        Dictionary<string, int> counts = new Dictionary<string, int>();

        // Extract the elements that occur odd times
        foreach (string word in words)
        {
            string wordInLowerCase = word.ToLower();

            if (counts.ContainsKey(wordInLowerCase))
            {
                counts[wordInLowerCase]++;
            }
            else
            {
                counts[wordInLowerCase] = 1;
            }
        }

        List<string> odds = new List<string>();

        foreach (KeyValuePair<string, int> entry in counts)
        {
            if (entry.Value % 2 == 1)
            {
                odds.Add(entry.Key);
            }
        }

        // Print result
        for (int i = 0; i < odds.Count; i++)
        {
            Console.Write(odds[i].ToString() + " ");
        }
    }
}
